presto-hive/src/main/java/com/facebook/presto/hive/HiveBasicStatistics.java

```diff
@@ -138,6 +138,23 @@ private static OptionalLong add(OptionalLong first, OptionalLong second)
         return OptionalLong.empty();
```

Comment:
do you have a test where you drop partitions from external table which did not have any statistics?

Reply:
Yes, a unit and an integration.